holidays/financial/national_stock_exchange_of_india.py

```diff
@@ -81,40 +115,45 @@ def _populate_public_holidays(self):
 
         # Hindu Calendar Holidays.
 
-        # Maha Shivaratri.
-        self._move_holiday(self._add_maha_shivaratri(tr("Maha Shivaratri")))
+        if self._year >= 2007:
+            # Maha Shivaratri.
+            self._move_holiday(self._add_maha_shivaratri(tr("Maha Shivaratri")))
 
         # Holi.
         self._move_holiday(self._add_holi(tr("Holi")))
 
-        # Ram Navami.
-        self._move_holiday(self._add_ram_navami(tr("Ram Navami")))
+        if self._year >= 2006:
+            # Ram Navami.
+            self._move_holiday(self._add_ram_navami(tr("Ram Navami")))
 
-        # Mahavir Jayanti.
-        self._move_holiday(self._add_mahavir_jayanti(tr("Mahavir Jayanti")))
+        if self._year >= 2006:
+            # Mahavir Jayanti.
+            self._move_holiday(self._add_mahavir_jayanti(tr("Mahavir Jayanti")))
 
         # Ganesh Chaturthi.
         self._move_holiday(self._add_ganesh_chaturthi(tr("Ganesh Chaturthi")))
 
         # Dussehra.
         self._move_holiday(self._add_dussehra(tr("Dussehra")))
 
-        # Diwali Laxmi Pujan.
-        self._move_holiday(self._add_gau_krida(tr("Diwali Laxmi Pujan")))
+        # Diwali Lakshmi Puja.
+        self._move_holiday(self._add_diwali_india(tr("Diwali Laxmi Pujan")))
 
-        # Diwali Balipratipada.
-        name = tr("Diwali Balipratipada")
-        # NSE's calendar adds an extra-day gap in 2022–2023
-        # between Diwali Laxmi Pujan and Diwali Balipratipada.
-        self._move_holiday(
-            self._add_bhai_dooj(name)
-            if self._year in {2022, 2023}
-            else self._add_govardhan_puja(name)
-        )
+        if self._year <= 2002 or self._year >= 2011:
+            # Diwali Balipratipada.
+            self._move_holiday(self._add_govardhan_puja(tr("Diwali Balipratipada")))
 
         # Guru Nanak Jayanti.
         self._move_holiday(self._add_guru_nanak_jayanti(tr("Guru Nanak Jayanti")))
 
+        if 2003 <= self._year <= 2010:
+            # Bhai Dooj.
+            self._move_holiday(self._add_bhai_dooj(tr("Bhai Bij")))
```

Comment:
```suggestion
            self._move_holiday(self._add_bhai_dooj(tr("Bhau Bhij")))
```
Sources contain different versions, I took the chronologically most recent one. Maybe I'm wrong.

Reply:
Hindi to english translation is opiniated, i changed from `Bhij` to `Bij` based on coderabbit suggestion. I think we should go with it.

Reply:
Make sence, will revert to name as per the document.